src/UnitOfWork.php

```diff
@@ -275,7 +300,7 @@ private function manageEntityReference($oid)
         $this->entityStateReferences[$id] = clone $entity;
     }
 
-    private function detectEntityChanges()
+    public function detectEntityChanges()
```

Comment:
If an onFlush listener decides to make changes to entities or make new entities it'll have to force the UnitOfWork to update it's changeset, similar to [computeChangeSet](http://doctrine-orm.readthedocs.io/projects/doctrine-orm/en/latest/reference/events.html#onflush) in doctrine-orm.
I would have made `computeChanges` and `computeRelationshipEntityChanges` public instead, but they require passing the original state as well, and there doesn't seem to be an analog for relationship references. I expect this will probably change when UnitOfWork is rewritten.